blockchain/vm/eips.go

```diff
@@ -117,6 +120,43 @@ func enableIstanbulComputationCostModification(jt *JumpTable) {
 	jt[SAR].computationCost = params.SarComputationCostIstanbul
 }
 
+// enable1153 applies EIP-1153 "Transient Storage"
+// - Adds TLOAD that reads from transient storage
+// - Adds TSTORE that writes to transient storage
+func enable1153(jt *JumpTable) {
+	jt[TLOAD] = &operation{
+		execute:         opTload,
+		constantGas:     params.WarmStorageReadCostEIP2929,
+		minStack:        minStack(1, 1),
+		maxStack:        maxStack(1, 1),
+		computationCost: params.TloadComputationCost,
+	}
+
+	jt[TSTORE] = &operation{
+		execute:         opTstore,
+		constantGas:     params.WarmStorageReadCostEIP2929,
+		minStack:        minStack(2, 0),
+		maxStack:        maxStack(2, 0),
+		computationCost: params.TstoreComputationCost,
+	}
+}
+
+// opTload implements TLOAD opcode
+func opTload(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
+	loc := scope.Stack.Peek()
+	val := evm.StateDB.GetTransientState(scope.Contract.Address(), common.BigToHash(loc))
+	loc.SetBytes(val.Bytes())
+	return nil, nil
+}
+
+// opTstore implements TSTORE opcode
+func opTstore(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
+	loc := scope.Stack.pop()
+	val := scope.Stack.pop()
+	evm.StateDB.SetTransientState(scope.Contract.Address(), common.BigToHash(loc), common.BigToHash(val))
```

Comment:
```suggestion
	evm.StateDB.SetTransientState(scope.Contract.Address(), common.BigToHash(loc), common.BigToHash(val))
	evm.interpreter.intPool.put(val)
```

Reply:
Can I ask why this change is needed?

Reply:
As we still use big integer pool for performance (I think the intpool stands for efficient memory usage), I suggest it be cached, but it's not necessary a work on this change.

Reply:
I see, I'll apply it. Thank you!

Reply:
Applied at https://github.com/klaytn/klaytn/pull/1988/commits/9fe8b33956ae04106cf25a226c1533ef032d0627

Reply:
Applied at https://github.com/klaytn/klaytn/pull/1988/commits/929848de661c8499b9368ab7f9144dc02d104e35.